packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanTreeOffset.js

```diff
@@ -14,33 +14,145 @@
-  const wrapperProps = hasChildren ? { onClick, role: 'switch', 'aria-checked': childrenVisible } : null;
-  const icon = hasChildren && (childrenVisible ? <IoIosArrowDown /> : <IoChevronRight />);
-  return (
-    <span className={`SpanTreeOffset ${hasChildren ? 'is-parent' : ''}`} {...wrapperProps}>
-      <span style={{ paddingLeft: `${level * 20}px` }} />
-      {icon && <span className="SpanTreeOffset--iconWrapper">{icon}</span>}
-    </span>
-  );
+type SpanTreeOffsetPropsType = SpanTreeOffsetOwnPropsType & {
+  hoverIndentGuideIds: Set<string>,
+};
+
+export class UnconnectedSpanTreeOffset extends React.PureComponent<SpanTreeOffsetPropsType> {
+  ancestorIds: string[];
+  props: SpanTreeOffsetPropsType;
+
+  static defaultProps = {
+    childrenVisible: false,
+    hasChildren: false,
+    onClick: null,
+  };
+
+  constructor(props: SpanTreeOffsetPropsType) {
+    super(props);
+
+    this.ancestorIds = [];
+    let currentSpan: Span = props.span;
+    while (currentSpan) {
+      currentSpan = _get(_find(currentSpan.references, { refType: 'CHILD_OF' }), 'span');
+      if (currentSpan) {
+        this.ancestorIds.push(currentSpan.spanID);
+      }
+    }
+
+    // Some traces have multiple root-level spans, this connects them all under one guideline and adds the
+    // necessary padding for the collapse icon on root-level spans.
+    this.ancestorIds.push('root');
+
+    this.ancestorIds.reverse();
+  }
+
+  /**
+   * If the mouse leaves to anywhere except another span with the same ancestor id, this span's ancestor id is
+   * removed from the set of hoverIndentGuideIds.
+   *
+   * @param {Object} event - React Synthetic event tied to mouseleave. Includes the related target which is
+   *     the element the user is now hovering.
+   * @param {string} ancestorId - The span id that the user was hovering over.
+   */
+  handleMouseLeave = (event: SyntheticMouseEvent<HTMLSpanElement>, ancestorId: string) => {
+    if (
+      !(event.relatedTarget instanceof HTMLSpanElement) ||
+      // !event.relatedTarget.getAttribute ||
+      event.relatedTarget.getAttribute('data--ancestor-id') !== ancestorId
+    ) {
+      this.props.removeHoverIndentGuideId(ancestorId);
+    }
+  };
+
+  /**
+   * If the mouse entered this span from anywhere except another span with the same ancestor id, this span's
+   * ancestorId is added to the set of hoverIndentGuideIds.
+   *
+   * @param {Object} event - React Synthetic event tied to mouseenter. Includes the related target which is
+   *     the last element the user was hovering.
+   * @param {string} ancestorId - The span id that the user is now hovering over.
+   */
+  handleMouseEnter = (
+    event: SyntheticMouseEvent<HTMLSpanElement> & { relatedTarget?: { getAttribute: string => string } },
+    ancestorId: string
+  ) => {
+    if (
+      !(event.relatedTarget instanceof HTMLSpanElement) ||
+      // !event.relatedTarget.getAttribute ||
```

Comment:
Extra LOC?

Reply:
I wasn't doing `instanceof` at first (rather `event.relatedTarget && event.relatedTarget.getAttribute && event.relatedTarget.getAttribute('data--ancestor-id') !== spanId`) but flow complained so I updated to the instance of check and then the second check was no longer necessary.
removed comment line from this handler and the other